module4/lessons/Week4/SecurityConcerns.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: page
+title: Security Concerns
+---
+
+## Learning Goals
+* Understand some security concerns of web development 
+* Implement hashed data storage
+* Implement secrets management
+
+## Data Security on the Web
+
+The bad news: the internet is a dangerous place.  Fortunately, we have some tools that developers can use to help lessen the danger of security breaches.  Today we are going to discuss some of the most common strategies developers use to protect data - this is not a complete list, but will get you thinking about the kinds of questions we need to answer when securing our applications!
+
+<section class='call-to-action' markdown='1'>
+
+### Gallery Walk
+
+We are going to create a Gallery of Security.  You will be assigned one of the topics listed below, and have 10 minutes to research on your own.  Then we will organize into groups, and your groups will have 20 minutes to create a slide (one single slide) that introduces this topic to the rest of the cohort.
+
+* Authentication/Authorization and API keys
+* API Rate Limiting
+* Overpost Prevention using Data Transfer Objects (DTOs)
```

Comment:
Might be useful to have an instructor note with couple sentence explanation of key points for these so that if the instructor isn't familiar they don't have to do independent research.

Reply:
Good call!